stardis/opacities/base.py

```diff
@@ -152,67 +137,80 @@ def calc_tau_photo(
     strength : float
         Coefficient to inverse cube term in equation for photoionization
         optical depth, expressed in cm^2.
-    cutoff_frequency : float
-        Minimum frequency of light to ionize atom.
 
     Returns
     -------
-    tau_photo : numpy.ndarray
+    alpha_photo : numpy.ndarray
         Array of shape (no_of_shells, no_of_frequencies). Photoiosnization
         optical depth in each shell for each frequency in tracing_nus.
     """
 
-    nl = splasma.level_number_density.loc[level] * marcs_model_fv.cell_length.values
+    fv_geometry = stellar_model.fv_geometry
+
+    ionization_energy = stellar_plasma.ionization_data.loc[(1, 1)]
 
-    tau_photo = np.zeros([len(marcs_model_fv), len(tracing_nus)])
+    alpha_h_photo = np.zeros([len(fv_geometry), len(tracing_nus)])
 
-    for i in range(len(tracing_nus)):
-        nu = tracing_nus[i]
-        if nu.value >= cutoff_frequency:
-            for j in range(len(marcs_model_fv)):
-                tau_photo[j, i] = strength * nl[j] * (cutoff_frequency / nu.value) ** 3
+    for level in levels:
+        alpha_h_photo_level = np.zeros([len(fv_geometry), len(tracing_nus)])
+        level_tuple = (1, 0, level)
+        cutoff_frequency = (
+            ionization_energy - stellar_plasma.excitation_energy.loc[level_tuple]
+        ) / const.h.cgs.value
+        n = stellar_plasma.level_number_density.loc[level_tuple]
+        for i in range(len(tracing_nus)):
+            nu = tracing_nus[i]
+            if nu.value >= cutoff_frequency:
+                for j in range(len(fv_geometry)):
+                    alpha_h_photo_level[j, i] = (
+                        strength * n[j] * (cutoff_frequency / nu.value) ** 3
+                    )
+        alpha_h_photo += alpha_h_photo_level
 
-    return tau_photo
+    return alpha_h_photo
 
 
 # line opacity
-def calc_tau_line(splasma, marcs_model_fv, tracing_nus):
+def calc_alpha_line_at_nu(stellar_plasma, stellar_model, tracing_nus):
     """
     Calculates line interaction optical depth.
 
     Parameters
     ----------
-    splasma : tardis.plasma.base.BasePlasma
+    stellar_plasma : tardis.plasma.base.BasePlasma
         Stellar plasma.
-    marcs_model_fv : pandas.core.frame.DataFrame
+    fv_geometry : pandas.core.frame.DataFrame
```

Comment:
But 2nd parameter is stellar_model

Reply:
This is fixed in a future PR. I know it should have been in the same one but I had other changes to make that were going to affect docstrings that I decided to combine them into one docstrings PR.